parsing/lexer.mll

```diff
@@ -418,35 +467,40 @@ rule token = parse
   | ".~"
       { error lexbuf
           (Reserved_sequence (".~", Some "is reserved for use in MetaOCaml")) }
-  | "~" raw_ident_escape (lowercase identchar * as name) ':'
-      { LABEL name }
-  | "~" (lowercase identchar * as name) ':'
+  | "~" (identstart identchar * as name) ':'
       { check_label_name lexbuf name;
         LABEL name }
-  | "~" (lowercase_latin1 identchar_latin1 * as name) ':'
-      { warn_latin1 lexbuf;
+  | "~" (raw_ident_escape? as escape) (ident_ext as raw_name) ':'
+      { let name = ident_for_extended lexbuf raw_name in
+        if escape="" then check_label_name lexbuf name;
```

Comment:
It's not immediately obvious how and why the check is different when `escape` is present. I suggest passing whether an `escape` is present to the `check_label_name` function. For example, `check_label_name ~has_escape:(escape<>"")`.
This way, the logic could be deduplicated from the OPTLABEL rule and documented in the check function.

Reply:
I took an alternative path, and added an `check_unicode_label_name` to factorize the two tests.

Reply:
`check_label_name` and `check_unicode_label_name` are practically identical. Perhaps they could be merged or shared more.

Reply:
Indeed, they are now merged together (with an optional argument).